service/header/p2p_exchange.go

```diff
@@ -8,48 +8,58 @@ import (
 	"github.com/libp2p/go-libp2p-core/network"
 	"github.com/libp2p/go-libp2p-core/peer"
 	"github.com/libp2p/go-libp2p-core/protocol"
-
 	tmbytes "github.com/tendermint/tendermint/libs/bytes"
 
-	pb "github.com/celestiaorg/celestia-node/service/header/pb"
 	"github.com/celestiaorg/go-libp2p-messenger/serde"
+
+	pb "github.com/celestiaorg/celestia-node/service/header/pb"
 )
 
 var exchangeProtocolID = protocol.ID("/header-exchange/v0.0.1")
 
-// exchange enables sending outbound ExtendedHeaderRequests as well as
+// P2PExchange enables sending outbound ExtendedHeaderRequests as well as
 // handling inbound ExtendedHeaderRequests.
-type exchange struct {
+type P2PExchange struct {
 	host host.Host
 	// TODO @renaynay: post-Devnet, we need to remove reliance of Exchange on one bootstrap peer
 	// Ref https://github.com/celestiaorg/celestia-node/issues/172#issuecomment-964306823.
-	peer  peer.ID
-	store Store
+	trustedPeer *peer.AddrInfo
+	store       Store
 
 	ctx    context.Context
 	cancel context.CancelFunc
 }
 
-func newExchange(host host.Host, peer peer.ID, store Store) *exchange {
-	return &exchange{
-		host:  host,
-		peer:  peer,
-		store: store,
+func NewP2PExchange(host host.Host, peer *peer.AddrInfo, store Store) *P2PExchange {
+	return &P2PExchange{
+		host:        host,
+		trustedPeer: peer,
+		store:       store,
 	}
 }
 
-func (ex *exchange) Start() {
+func (ex *P2PExchange) Start(ctx context.Context) error {
+	if ex.trustedPeer != nil {
+		err := ex.host.Connect(ctx, *ex.trustedPeer)
+		if err != nil {
+			log.Errorw("connecting to trusted peer", "err", err)
+			log.Warn("HEADERS WONT BE SYNCHRONIZED - PLEASE RESTART WITH TRUSTED PEER BEING ONLINE")
```

Comment:
```suggestion
			log.Warn("HEADERS WONT BE SYNCHRONIZED - PLEASE RESTART WITH TRUSTED ACTIVE PEER")
```

Reply:
I think these are the changes actually introduced in #220 btw?

Reply:
Yeah. This PR says in description that it is based on #220

Reply:
are you gonna commit this suggestion or no